src/careamics/careamist.py

```diff
@@ -596,9 +617,265 @@ def predict(
+        configuration parameters will be used, with the `patch_size` instead of
+        `tile_size`.
+
+        Test-time augmentation (TTA) can be switched off using the `tta_transforms`
+        parameter. The TTA augmentation applies all possible flip and 90 degrees
+        rotations to the prediction input and averages the predictions. TTA augmentation
+        should not be used if you did not train with these augmentations.
+
+        Note that if you are using a UNet model and tiling, the tile size must be
+        divisible in every dimension by 2**d, where d is the depth of the model. This
+        avoids artefacts arising from the broken shift invariance induced by the
+        pooling layers of the UNet. If your image has less dimensions, as it may
+        happen in the Z dimension, consider padding your image.
+
+        Parameters
+        ----------
+        source : CAREamicsPredData, pathlib.Path or str
+            Data to predict on.
+        batch_size : int, default=1
+            Batch size for prediction.
+        tile_size : tuple of int, optional
+            Size of the tiles to use for prediction.
+        tile_overlap : tuple of int, default=(48, 48)
+            Overlap between tiles.
+        axes : str, optional
+            Axes of the input data, by default None.
+        data_type : {"array", "tiff", "custom"}, optional
+            Type of the input data.
+        tta_transforms : bool, default=True
+            Whether to apply test-time augmentation.
+        dataloader_params : dict, optional
+            Parameters to pass to the dataloader.
+        read_source_func : Callable, optional
+            Function to read the source data.
+        extension_filter : str, default=""
+            Filter for the file extension.
+        write_type : {"tiff", "custom"}, default="tiff"
+            The data type to save as, includes custom.
+        write_extension : str, optional
+            If a known `write_type` is selected this argument is ignored. For a custom
+            `write_type` an extension to save the data with must be passed.
+        write_func : WriteFunc, optional
+            If a known `write_type` is selected this argument is ignored. For a custom
+            `write_type` a function to save the data must be passed. See notes below.
+        write_func_kwargs : dict of {str: any}, optional
+            Additional keyword arguments to be passed to the save function.
+        **kwargs : Any
+            Unused.
+
+        Raises
+        ------
+        ValueError
+            If mean and std are not provided in the configuration.
+        ValueError
+            If tile size is not divisible by 2**depth for UNet models.
+        ValueError
+            If tile overlap is not specified.
+        """
+        # create and set correct write strategy
+        tiled = tile_size is not None
+        write_strategy = create_write_strategy(
```

Comment:
Could it be a bit more concise to pack the write strategy selection into the prediction_writer itself, something like `self.prediction_writer.set_writing_strategy(write_type, ...)` and create the `write_strategy` there directly?

Reply:
This was a deliberate design choice to reduce coupling between `PredictionWriterCallback` and the `WriteStrategy` classes. Removing all responsibility of `WriteStrategy` construction from `PredictionWriterCallback` means that the write strategies can be almost arbitrarily altered, with the constraint that they have a `write_batch` method, without having to alter the `PredictionWriterCallback`.

Reply:
Ok I did break the pattern rules with the `from_write_func_params` though 😅. I put this in for convenience of users of the lightning API, but I am now considering removing though. The issue is that the `create_write_strategy` might have to change a bit because I don't know the parameters that the `WriteZarrTiles` strategy will need for initialisation. This means the arguments for `from_write_func_params` will also need to change and also those of a potential `set_write_strategy` method.